see https://bugs.gentoo.org/966047
PR pending https://github.com/jkriege2/JKQtPlotter/pull/156.patch
avoid Qt5 automagic, respect an exported value of QT_VERSION_MAJOR
--- a/cmake/jkqtplotter_common_qtsettings.cmake
+++ b/cmake/jkqtplotter_common_qtsettings.cmake
@@ -4,7 +4,11 @@ set(CMAKE_AUTOMOC ON)
 set(CMAKE_AUTORCC ON)
 set(CMAKE_AUTOUIC ON)
 
-find_package(QT NAMES Qt6 Qt5 COMPONENTS Core REQUIRED)
+if(QT_VERSION_MAJOR)
+    find_package(QT NAMES Qt${QT_VERSION_MAJOR} COMPONENTS Core REQUIRED)
+else()
+    find_package(QT NAMES Qt6 Qt5 COMPONENTS Core REQUIRED)
+endif()
 
 if(QT_VERSION_MAJOR LESS 5)
     message(FATAL_ERROR "Minimum supported Qt version is 5, but you are trying to compile against Qt${QT_VERSION_MAJOR}")
